src/main/java/io/jenkins/plugins/checks/github/GitHubChecksPublisherFactory.java

```diff
@@ -37,42 +45,67 @@ public GitHubChecksPublisherFactory() {
 
     @Override
     protected Optional<ChecksPublisher> createPublisher(final Run<?, ?> run, final TaskListener listener) {
-        return createPublisher(run, DisplayURLProvider.get().getRunURL(run), listener);
-    }
-
-    @VisibleForTesting
-    Optional<ChecksPublisher> createPublisher(final Run<?, ?> run, final String runURL, final TaskListener listener) {
-        PluginLogger logger = createLogger(getListener(listener));
-        
-        GitSCMChecksContext gitSCMContext = new GitSCMChecksContext(run, runURL);
-        if (gitSCMContext.isValid(logger)) {
-            return Optional.of(new GitHubChecksPublisher(gitSCMContext, getListener(listener)));
+        try {
+            return createPublisher(run, DisplayURLProvider.get().getRunURL(run), listener);
+        }
+        catch (UnsupportedEncodingException e) {
+            LOGGER.log(Level.WARNING, "Could not create logger.", e);
         }
 
-        return createPublisher(listener, logger, new GitHubSCMSourceChecksContext(run, runURL, scmFacade));
+        return Optional.empty();
     }
 
     @Override
     protected Optional<ChecksPublisher> createPublisher(final Job<?, ?> job, final TaskListener listener) {
-        return createPublisher(job, DisplayURLProvider.get().getJobURL(job), listener);
+        try {
+            return createPublisher(job, DisplayURLProvider.get().getJobURL(job), listener);
+        }
+        catch (UnsupportedEncodingException e) {
+            LOGGER.log(Level.WARNING, "Could not create logger.", e);
+        }
+
+        return Optional.empty();
     }
 
     @VisibleForTesting
-    Optional<ChecksPublisher> createPublisher(final Job<?, ?> job, final String jobURL, final TaskListener listener) {
-        PluginLogger logger = createLogger(getListener(listener));
+    Optional<ChecksPublisher> createPublisher(final Run<?, ?> run, final String runURL, final TaskListener listener)
+            throws UnsupportedEncodingException {
+        ByteArrayOutputStream cause = new ByteArrayOutputStream();
+        try (PrintStream ps = new PrintStream(cause, true, StandardCharsets.UTF_8.name())) {
```

Comment:
Why are you creating a new `PrintStream`? Can't you use the `TaskListener` directly? See line 118.

Reply:
We can either place the GitHubSCM first (this has the same effect for GitSCM users) or we can use the `FilteredLog` for both context types and print the log only in case that both return nothing.

Reply:
yes, I'm finding such a logger, I'll use `FilteredLog`.